### PR TITLE
Role based ACL migration

### PR DESCRIPTION
**Pull request by howawong**
1. conversation class updated
2. conversation_handlers added
3. schema, user, roles classes added
4. user_conversation updated
5. conversation related utils function added
6. participant_ids and admin_ids removed from conversation record
7. is_admin boolean added to user conversation record

### CODE REVIEW

**Comment by howawong** [comment]
Connects #109

File: chat/user_conversation.py [hunk 1]
```diff
@@ -4,58 +4,61 @@
...
+from .query import Query
+from .utils import _get_container, _get_schema_name
 
 
-class UserConversation():
-    def __init__(self, conversation, participant_id, master_key=None):
-        if master_key is None:
-            master_key = skyoptions.masterkey
+class UserConversation(Record):
+    def __init__(self, conversation, user):
```
Comment (rickmak) [inline chat/user_conversation.py]: conversation -> conversation_id if we accept id

File: chat/conversation_handlers.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+    uc = UserConversation(conversation_id, current_user_id())
+    uc.delete()
+    return {'status': 'OK'}
+
+
+def handle_add_participants(conversation_id,
+                            participants,
+                            is_first_time=False):
+
+    r = _get_conversation(conversation_id)
```
Comment (howawong) [inline chat/conversation_handlers.py]: Okay. That's a mistake.

File: chat/user_conversation.py (hunk 1 above)
Comment (howawong) [inline chat/user_conversation.py]: Yes, the naming was bad. I will change it.

**Pull request merged**